Mathlib/Data/Matroid/Circuit.lean

```diff
@@ -260,10 +262,34 @@ lemma Circuit.eq_fundCircuit_of_subset (hC : M.Circuit C) (hI : M.Indep I) (hCss
     · rw [hI.mem_closure_iff]
       exact .inl (hC.dep.superset hCss (insert_subset (hC.subset_ground heC) hI.subset_ground))
     exact hC.not_indep (hI.subset (hCss.trans (by simp [heI])))
-  refine insert_subset heC (inter_subset_right.trans ?_)
-  refine (sInter_subset_of_mem (t := C \ {e}) ?_).trans diff_subset
+  rw [fundCircuit_eq_sInter
+    (M.closure_subset_closure hCeI <| hC.mem_closure_diff_singleton_of_mem heC)]
+  refine insert_subset heC <| (sInter_subset_of_mem (t := C \ {e}) ?_).trans diff_subset
   simp [hCss, hC.mem_closure_diff_singleton_of_mem heC]
 
+lemma fundCircuit_restrict {R : Set α} (hIR : I ⊆ R) (heR : e ∈ R) (hR : R ⊆ M.E) :
+    (M ↾ R).fundCircuit e I = M.fundCircuit e I := by
+  simp_rw [fundCircuit, M.restrict_closure_eq (R := R) (X := {e}) (by simpa)]
+  refine subset_antisymm (insert_subset_insert (inter_subset_inter_right _ ?_))
+    (insert_subset_insert (inter_subset_inter_right _ ?_))
+  · refine subset_sInter fun J ⟨hJI, heJ⟩ ↦ sInter_subset_of_mem ⟨hJI, ?_⟩
+    simp only [restrict_closure_eq', union_subset_iff, subset_union_right, and_true]
+    refine (inter_subset_inter_left _ ?_).trans subset_union_left
+    rwa [inter_eq_self_of_subset_left (hJI.trans hIR)]
+  refine subset_sInter fun J ⟨hJI, heJ⟩ ↦ sInter_subset_of_mem
+    ⟨hJI, M.closure_subset_closure_of_subset_closure ?_⟩
+  rw [restrict_closure_eq _ (hJI.trans hIR) hR] at heJ
+  simp only [subset_inter_iff, inter_subset_right, and_true] at heJ
+  exact subset_trans (by simpa [M.mem_closure_of_mem' (mem_singleton e) (hR heR)]) heJ
+
+@[simp] lemma fundCircuit_restrict_univ (M : Matroid α) :
+    (M ↾ univ).fundCircuit e I = M.fundCircuit e I := by
```

Comment:
I would have guessed that `M ↾ univ = M`; and if true that should be simp and would imply this one... Why's that false?

Reply:
Oh, I see - restricting to the ground set has this property instead.